cloudinit/config/cc_wireguard.py

```diff
@@ -0,0 +1,290 @@
+# Author: Fabian Lichtenegger-Lukas <fabian.lichtenegger-lukas@nts.eu>
+# Author: Josef Tschiggerl <josef.tschiggerl@nts.eu>
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""Wireguard"""
+import re
+from textwrap import dedent
+
+from cloudinit import log as logging
+from cloudinit import subp, util
+from cloudinit.cloud import Cloud
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.settings import PER_INSTANCE
+
+MODULE_DESCRIPTION = dedent(
+    """\
+Wireguard module provides a dynamic interface for configuring
+Wireguard (as a peer or server) in an easy way.
+
+This module takes care of..
+  - writing interface configuration files
+  - enabling and starting interfaces
+  - installing wireguard-tools package
+  - loading wireguard kernel module
+  - executing readiness probes
+
+What's a readiness probe?
+The idea behind readiness probes is to ensure Wireguard connectivity
+before continuing the cloud-init process. This could be useful if you
+need access to specific services like an internal APT Repository Server
+(e.g Landscape) to install/update packages.
+
+Example:
+An edge device can't access the internet but uses cloud-init modules that
+will install packages (e.g landscape, packages, ubuntu_advantage). Those
+modules will fail due to missing internet connection. The "wireguard" module
+fixes that problem as it waits until all readinessprobes (which can be
+arbitrary commands - e.g. checking if a proxy server is reachable over
+Wireguard network) are finished before continuing the cloud-init
+"config" stage.
+"""
+)
+
+meta: MetaSchema = {
+    "id": "cc_wireguard",
+    "name": "Wireguard",
+    "title": "Module to configure Wireguard tunnel",
+    "description": MODULE_DESCRIPTION,
+    "distros": ["ubuntu"],
+    "frequency": PER_INSTANCE,
```

Comment:
We have introduced a new schema property, in https://github.com/canonical/cloud-init/pull/1597, to skip the execution of a module if some config keys are not present. And I think this module can be skipped if `wireguard` is not given in the cloud config, right? If so, could you please rebase main and apply:

```suggestion
    "frequency": PER_INSTANCE,
    "activate_by_schema_keys": ["wireguard"],
```